javascript/index.js

```diff
@@ -25,16 +26,116 @@
+        getInstruction('mashedPotatoes',4, (step4) => {
+          document.querySelector("#mashedPotatoes").innerHTML += `<li>${step4}</li>`
+          getInstruction('mashedPotatoes',4, (step4) => {
+            document.querySelector("#mashedPotatoes").innerHTML += `<li>Mashed potatoes are ready!</li>`
+            document.querySelector("#mashedPotatoesImg").removeAttribute("hidden");
+          }, (err) => console.log(err));
+        }, (err) => console.log(err));
+      }, (err) => console.log(err));
+    }, (err) => console.log(err));
+  }, (err) => console.log(err));
+}, (err) => console.log(err));
 
 // Iteration 2 - using promises
-// ...
+obtainInstruction('steak', 0)
+  .then( (step0) => {
+    document.querySelector("#steak").innerHTML += `<li>${step0}</li>`
+    //  ... Your code here
+    return obtainInstruction('steak', 1);
+  })
+  .then( (step1) => {
+    document.querySelector("#steak").innerHTML += `<li>${step1}</li>`
+    return obtainInstruction('steak', 2);
+  })
+  .then( (step2) => {
+    document.querySelector("#steak").innerHTML += `<li>${step2}</li>`
+    return obtainInstruction('steak', 3);
+  })
+  .then( (step3) => {
+    document.querySelector("#steak").innerHTML += `<li>${step3}</li>`
+    return obtainInstruction('steak', 4);
+  })
+  .then( (step4) => {
+    document.querySelector("#steak").innerHTML += `<li>${step4}</li>`
+    return obtainInstruction('steak', 5);
+  })
+  .then( (step5) => {
+    document.querySelector("#steak").innerHTML += `<li>${step5}</li>`
+    return obtainInstruction('steak', 6);
+  })
+  .then( (step6) => {
+    document.querySelector("#steak").innerHTML += `<li>${step6}</li>`
+    return obtainInstruction('steak', 7);
+  })
+  .then( (step7) => {
+    document.querySelector("#steak").innerHTML += `<li>Steak is ready!</li>`
+    document.querySelector("#steakImg").removeAttribute("hidden");
+    return obtainInstruction('steak', 7);
+  });
 
 // Iteration 3 using async/await
-// ...
+
+async function makeBroccoli(){
+  try{
+  const step0 = await obtainInstruction("broccoli", 0);
+  document.querySelector("#broccoli").innerHTML += `<li>${step0}</li>`;
+  const step1 = await obtainInstruction("broccoli", 1);
+  document.querySelector("#broccoli").innerHTML += `<li>${step1}</li>`;
+  const step2 = await obtainInstruction("broccoli", 2);
+  document.querySelector("#broccoli").innerHTML += `<li>${step2}</li>`;
+  const step3 = await obtainInstruction("broccoli", 3);
+  document.querySelector("#broccoli").innerHTML += `<li>${step3}</li>`;
+  const step4 = await obtainInstruction("broccoli", 4);
+  document.querySelector("#broccoli").innerHTML += `<li>${step4}</li>`;
+  const step5 = await obtainInstruction("broccoli", 5);
+  document.querySelector("#broccoli").innerHTML += `<li>${step5}</li>`;
+  const step6 = await obtainInstruction("broccoli", 6);
+  document.querySelector("#broccoli").innerHTML += `<li>${step6}</li>`;
+  const step7 = await obtainInstruction("broccoli", 6);
```

Comment:
Here, you didn't need this `step7` variable, as you could do the steps bellow chained to the previous step.